src/middlewared/middlewared/plugins/service.py

```diff
@@ -405,6 +405,16 @@ async def identify_process(self, procname):
                 if await service.identify(procname):
                     return service_name
 
+    @private
+    async def get_unit_state(self, service):
+        service_object = await self.middleware.call('service.object', service)
+        return await service_object.get_unit_state()
+
+    @private
+    async def failover(self, service):
```

Comment:
It would probably be less race-prone if we directly passed in argument for new failover status.